examples/factor_analysis/latent_factor_from_htfa.py

```diff
@@ -0,0 +1,77 @@
+#  Copyright 2016 Intel Corporation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import scipy.io
+from scipy.stats import stats
+import numpy as np
+import nibabel as nib
+from nilearn.input_data import NiftiMasker
+import os
+from mpi4py import MPI
+
+comm = MPI.COMM_WORLD
+rank = comm.Get_rank()
+size = comm.Get_size()
+if rank == 0:
+    import logging
+    import sys
+    logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+
+n_subj = 2
+data_dir = os.getcwd() + '/data'
+url = []
+url.append(' https://www.dropbox.com/s/r5s9tg4ekxzbrco/s0.mat?dl=0')
+url.append(' https://www.dropbox.com/s/39tr01m76vxwaqa/s1.mat?dl=0')
+
+#get fMRI data and scanner RAS coordinates
+data = []
+R = []
+for idx in range(n_subj):
+    if idx % size == rank:
+        #download data
+        file_name = os.path.join(data_dir, 's' + str(idx) + '.mat')
+        cmd = 'curl --location --create-dirs -o ' + file_name + url[idx]
```

Comment:
I think there is a race condition here. Multiple ranks will attempt to create the directory at the same time. I actually saw this error triggered by the `curl` from rank 1, which resulted in `s1.mat` not being downloaded:
`Error creating directory /home/mihai/Surse/brainiak/examples/factor_analysis/data.`
You should attempt to create the directory only once, in rank 0.